services/brig/src/Brig/SMTP.hs

```diff
@@ -72,18 +143,106 @@ initSMTP lg host port credentials connType = do
           SMTP.connectSMTPSSLWithSettings (unpack host) $
             SMTP.defaultSettingsSMTPSSL {SMTP.sslPort = p}
       ok <- case credentials of
-        (Just (Username u, Password p)) -> SMTP.authenticate SMTP.LOGIN (unpack u) (unpack p) conn
+        (Just (Username u, Password p)) ->
+          ensureTimeout $
+            SMTP.authenticate SMTP.LOGIN (unpack u) (unpack p) conn
         _ -> pure True
-      pure (ok, conn)
-    create = do
-      (ok, conn) <- connect
       if ok
-        then Logger.log lg Logger.Debug (msg $ val "Established connection to: " +++ host)
-        else Logger.log lg Logger.Warn (msg $ val "Failed to established connection, check your credentials to connect to: " +++ host)
-      pure conn
-    destroy c = do
-      SMTP.closeSMTP c
-      Logger.log lg Logger.Debug (msg $ val "Closing connection to: " +++ host)
-
-sendMail :: MonadIO m => SMTP -> Mail -> m ()
-sendMail s m = liftIO $ withResource (s ^. pool) $ SMTP.sendMail m
+        then pure conn
+        else CE.throw SMTPUnauthorized
+
+    create :: IO SMTP.SMTPConnection
+    create =
+      logExceptionOrResult
+        lg
+        ("Creating pooled SMTP connection to " ++ unpack host)
+        establishConnection
+
+    destroy :: SMTP.SMTPConnection -> IO ()
+    destroy c =
+      logExceptionOrResult lg ("Closing pooled SMTP connection to " ++ unpack host) $
+        (ensureTimeout . SMTP.gracefullyCloseSMTP) c
+
+logExceptionOrResult :: (MonadIO m, MonadCatch m) => Logger -> String -> m a -> m a
+logExceptionOrResult lg actionString action = do
+  let msg' = msg ("SMTP connection result" :: String)
+  res <-
+    catches
+      action
+      [ Handler
+          ( \(e :: SMTPPoolException) -> do
+              let resultLog = case e of
+                    SMTPUnauthorized ->
+                      ("Failed to establish connection, check your credentials." :: String)
+                    SMTPConnectionTimeout -> ("Connection timeout." :: String)
```

Comment:
```suggestion
              let resultLog = show e
```

?  not sure about this one :)

Reply:
I'd argue that it's fine to be a bit more chatty in log messages. Finally, they are meant to be interpreted by Ops people outside of Wire.